flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java

```diff
@@ -738,13 +738,17 @@ protected void validateTableSource(TableSource<?> tableSource) {
 		TableSourceValidation.validateTableSource(tableSource, tableSource.getTableSchema());
 	}
 
-	protected List<Transformation<?>> translate() {
-		List<Transformation<?>> transformations = translate(bufferedModifyOperations);
-		bufferedModifyOperations.clear();
+	protected List<Transformation<?>> translateAndClearBuffer() {
+		List<Transformation<?>> transformations;
+		try {
+			transformations = translate(bufferedModifyOperations);
+		} finally {
```

Comment:
Could you add a comment on this that why do we force to clear buffers even if the `translate()` failed?